File: app/src/test/java/in/testpress/testpress/util/EmailValidatorTest.java
package in.testpress.testpress.util;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.JUnit4;
import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;

@RunWith(JUnit4.class)
public class EmailValidatorTest {

    @Test public void emailValidator_CorrectEmailSimple_ReturnsTrue() {
        assertTrue(Validator.isEmailValid("name@email.com"));
    }

    @Test
    public void emailValidator_CorrectEmailSubDomain_ReturnsTrue() {
        assertTrue(Validator.isEmailValid("name@email.co.uk"));
    }

    @Test
    public void emailValidator_InvalidEmailNoTld_ReturnsFalse() {
        assertFalse(Validator.isEmailValid("name@email"));
    }

    @Test
    public void emailValidator_InvalidEmailDoubleDot_ReturnsFalse() {
        assertFalse(Validator.isEmailValid("name@email..com"));
    }

    @Test
    public void emailValidator_InvalidEmailNoUsername_ReturnsFalse() {
        assertFalse(Validator.isEmailValid("@email.com"));
    }

    @Test
    public void emailValidator_EmptyString_ReturnsFalse() {
        assertFalse(Validator.isEmailValid(""));
    }

    @Test
    public void emailValidator_NullEmail_ReturnsFalse() {
        assertFalse(Validator.isEmailValid(null));
    }
}

File: app/src/test/java/in/testpress/testpress/util/EmailValidatorTest.kt
package `in`.testpress.testpress.util

import `in`.testpress.testpress.util.Validator.isEmailValid
import org.junit.Assert
import org.junit.Test
import org.junit.runner.RunWith
import org.junit.runners.JUnit4

@RunWith(JUnit4::class)
class EmailValidatorTest {
    @Test
    fun correctEmailReturnsTrue() {
        Assert.assertTrue(isEmailValid("name@email.com"))
    }

    @Test
    fun emailWithSubDomainReturnsTrue() {
        Assert.assertTrue(isEmailValid("name@email.co.uk"))
    }

    @Test
    fun emailWithNoTLDReturnsFalse() {
        Assert.assertFalse(isEmailValid("name@email"))
    }

    @Test
    fun emailWithDoubleDotReturnsFalse() {
        Assert.assertFalse(isEmailValid("name@email..com"))
    }

    @Test
    fun emailWithNoUsernameReturnsFalse() {
        Assert.assertFalse(isEmailValid("@email.com"))
    }

    @Test
    fun emptyStringReturnsFalse() {
        Assert.assertFalse(isEmailValid(""))
    }

    @Test
    fun nullReturnsFalse() {
        Assert.assertFalse(isEmailValid(null))
    }
}
